Dockerfile

```diff
@@ -39,14 +59,17 @@ RUN GRPC_HEALTH_PROBE_VERSION=v0.4.15 && \
     curl -fL -o /tools/grpc_health_probe https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/${GRPC_HEALTH_PROBE_VERSION}/grpc_health_probe-${TARGETOS}-${TARGETARCH} && \
     chmod +x /tools/grpc_health_probe
 
-
+####################################################################################################
+# final
+####################################################################################################
 FROM ghcr.io/akuity/bookkeeper:v0.1.0-rc.19 as final
 
 USER root
 
 COPY --from=builder /kargo/bin/ /usr/local/bin/
 COPY --from=tools /tools/ /usr/local/bin/
+COPY --from=ui-builder /ui/build /ui/build
 
-USER nonroot
+USER 1000:0
```

Comment:
I'm not sure what the consequences of this change may be.

I don't believe a user with this ID exists in the image.

Reply:
This change enables us to improve our install manifests with `securityContext` that is the most compatible across both OpenShift and vanilla Kubernetes.

It's preferable to use a **_numeric_** USER in a Dockerfile (instead of a named user) because it makes it possible to leverage the `runAsNonRoot: true` option in the Pod securityContext without other extra options.

```yaml
          securityContext:
            runAsNonRoot: true
```

When using numeric USER, Kubernetes is able to infer from the image layers that the container indeed does not run as root because it can clearly see a number instead of string after the `USER` directive. A numeric USER in the Dockerfile (combined with `runAsNonRoot: true`) is also preferable to using `runAsUser: 1000` option in the Pod:

```yaml
  securityContext:
    runAsUser: 1000
```

which IIRC, causes problems with OpenShift which I believe has security constraints that enforces that pods run within a specific numeric UID range. Somehow they can do UID reassignment if the USER is defined in the Dockerfile, but will not do it if `runAsUser: 1000` is used in the manifests.